pandas/core/groupby/generic.py

```diff
@@ -1805,21 +1805,24 @@ def value_counts(
                 name = self._selected_obj.name
                 keys = [] if name in in_axis_names else [self._selected_obj]
             else:
+                if subset is not None:
```

Comment:
2 things: What happens if a column from subset is not in the DataFrame at all?
Can we add a test with duplicated column names? Shouldn't be an issue with your implementation, but should help in avoiding potential regressions

Reply:
> What happens if a column from subset is not in the DataFrame at all?

No exception would be raised & would act like there was no subset. Changed this to raising a `ValueError`.

> Can we add a test with duplicated column names? Shouldn't be an issue with your implementation, but should help in avoiding potential regressions

Added

Reply:
Thx